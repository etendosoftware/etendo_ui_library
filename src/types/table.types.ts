import { DimensionValue } from 'react-native';

export type Metadata = {
  primary?: boolean;
  visible?: boolean;
  label?: string;
  key?: string;
  type?: 'number' | 'string' | 'boolean' | 'date';
  width?: DimensionValue;
  title?: boolean;
};
