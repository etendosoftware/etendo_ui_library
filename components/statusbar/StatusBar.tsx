import React from 'react';

import {StatusBarProps} from './StatusBar.types';
import {StatusBarStep1Active} from './StatusBarStep1Active';
import {StatusBarStep2Active} from './StatusBarStep2Active';
import {StatusBarStep3Active} from './StatusBarStep3Active';

export const StatusBar = ({data, step}: StatusBarProps) => {
  if (step <= 1 || step > 3) {
    return <StatusBarStep1Active data={data} step={step} />;
  } else if (step === 2) {
    return <StatusBarStep2Active data={data} step={step} />;
  } else {
    return <StatusBarStep3Active data={data} step={step} />;
  }
};
