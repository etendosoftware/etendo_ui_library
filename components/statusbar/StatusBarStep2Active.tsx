import React from 'react';
import {View, Text, Image} from 'react-native';
import {StatusBarStyleVariant} from './StatusBar.styles';
import {StatusBarProps} from './StatusBar.types';

export const StatusBarStep2Active = ({data}: StatusBarProps) => {
  return (
    <View style={StatusBarStyleVariant.primary.containerStatusBar}>
      <View
        style={StatusBarStyleVariant.primary.containerStatusBarMiddleBlueTop}
      />
      <View
        style={
          StatusBarStyleVariant.primary.containerStatusBarMiddleYellowBottom
        }
      />

      <View
        style={
          StatusBarStyleVariant.primary
            .containerStatusBarMiddleOfMiddleHorizontalYellowTop
        }
      />

      <View
        style={
          StatusBarStyleVariant.primary
            .containerStatusBarMiddleOfMiddleHorizontalYellowBottom
        }
      />

      <View
        style={
          StatusBarStyleVariant.primary
            .containerStatusBarMiddleOfMiddleHorizontalBlueTop
        }
      />

      <View style={StatusBarStyleVariant.primary.containerItemDisabledStep2}>
        <Text
          style={StatusBarStyleVariant.primary.containerItemDisabledStep1Text}
        >
          {data[0].step}
        </Text>
      </View>
      <View style={StatusBarStyleVariant.primary.containerItemActiveStep2}>
        <View style={StatusBarStyleVariant.primary.activeItemStep2LeftSection}>
          <Image
            source={data[1].activeImage}
            style={StatusBarStyleVariant.primary.step2Image}
          />
          <View style={StatusBarStyleVariant.primary.activeItemStep2Text}>
            <Text style={StatusBarStyleVariant.primary.activeItemStep2TextTop}>
              {data[1].activeTextTop}
            </Text>
            <Text
              style={StatusBarStyleVariant.primary.activeItemStep2TextBottom}
            >
              {data[1].activeTextBottom}
            </Text>
          </View>
        </View>
        <Text style={StatusBarStyleVariant.primary.activeItemStep2TextRight}>
          {data[1].step}
        </Text>
      </View>
      <View
        style={StatusBarStyleVariant.primary.containerItemDisabledStep3Right}
      >
        <Text
          style={
            StatusBarStyleVariant.primary.containerItemDisabledStep3TextRight
          }
        >
          {data[2].step}
        </Text>
      </View>
    </View>
  );
};
