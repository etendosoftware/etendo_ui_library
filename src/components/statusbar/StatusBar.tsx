/* Imports */
import React from 'react';
import { ViewEtendo } from '../viewEtendo/ViewEtendo';
import { StatusBarStyleVariant } from './StatusBar.styles';
import { StatusBarProps } from './StatusBar.types';
import { StepActive } from './StepActive';
import { StepDisabled } from './StepDisabled';

const StatusBar = ({ data, step }: StatusBarProps) => {
  return (
    <ViewEtendo style={StatusBarStyleVariant.primary.statusBarContainer}>
      {data.map((item, idx) => {
        if (idx === step - 1) {
          return <StepActive item={item} step={step} key={'status1' + idx} />;
        } else {
          return <StepDisabled item={item} step={step} key={'status2' + idx} />;
        }
      })}
    </ViewEtendo>
  );
};

export default StatusBar;
