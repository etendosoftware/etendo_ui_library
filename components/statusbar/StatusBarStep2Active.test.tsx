import React from 'react';
import renderer from 'react-test-renderer';

import {Info} from './StatusBar.types';
import {StatusBarStep2Active} from './StatusBarStep2Active';

const dataset: Info[] = [
  {
    step: '1. Vehicles',
    activeImage: require('../../assets/images/icons/calendar-secundary.png'),
    activeTextTop: 'Current plan date',
    activeTextBottom: '03 January, 2022',
  },
  {
    step: '2. Visits',
    activeImage: require('../../assets/images/icons/calendar-secundary.png'),
    activeTextTop: 'Current plan date',
    activeTextBottom: '03 January, 2022',
  },
  {
    step: '3. Routes',
    activeImage: require('../../assets/images/icons/calendar-secundary.png'),
    activeTextTop: 'Current plan date',
    activeTextBottom: '03 January, 2022',
  },
];

describe('Running Test for StatusBarStep1Active', () => {
  it('Check StatusBarStep1Active', () => {
    const tree = renderer
      .create(<StatusBarStep2Active data={dataset} step={1} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
