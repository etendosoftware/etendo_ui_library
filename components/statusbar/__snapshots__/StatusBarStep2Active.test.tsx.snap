// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Running Test for StatusBarStep1Active Check StatusBarStep1Active 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FAD614",
      "flex": 1,
      "height": 45,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#202452",
        "flex": 1,
        "height": "50%",
        "width": "100%",
      }
    }
  />
  <View
    style={
      Object {
        "backgroundColor": "#FAD614",
        "flex": 1,
        "height": "50%",
        "width": "100%",
      }
    }
  />
  <View
    style={
      Object {
        "backgroundColor": "#FAD614",
        "height": "50%",
        "position": "absolute",
        "width": "50%",
      }
    }
  />
  <View
    style={
      Object {
        "backgroundColor": "#202452",
        "height": "50%",
        "width": "50%",
      }
    }
  />
  <View
    style={
      Object {
        "backgroundColor": "#202452",
        "borderRadius": 26,
        "height": "50%",
        "left": "80%",
        "position": "absolute",
        "width": "20%",
        "zIndex": 1,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#202452",
        "borderTopEndRadius": 26,
        "borderTopRightRadius": 26,
        "flex": 1,
        "flexDirection": "row",
        "height": "100%",
        "justifyContent": "center",
        "position": "absolute",
        "width": "40%",
        "zIndex": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#FFFFFF",
          "fontSize": 17,
          "fontWeight": "bold",
          "marginLeft": 265,
        }
      }
    >
      1. Vehicles
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FAD614",
        "borderBottomLeftRadius": 26,
        "borderTopEndRadius": 26,
        "borderTopRightRadius": 26,
        "flex": 1,
        "flexDirection": "row",
        "height": "100%",
        "justifyContent": "center",
        "left": "40%",
        "position": "absolute",
        "width": "40.4%",
        "zIndex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "height": "80%",
          "left": 0,
          "marginRight": 45,
          "width": "35%",
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../assets/images/icons/calendar-secundary.png",
          }
        }
        style={
          Object {
            "alignItems": "center",
            "height": 28,
            "marginTop": "4px",
            "width": 28,
          }
        }
      />
      <View
        style={
          Object {
            "height": 100,
            "marginLeft": 15,
            "width": 250,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#202452",
              "fontSize": 14,
              "fontWeight": "bold",
              "marginTop": 4,
            }
          }
        >
          Current plan date
        </Text>
        <Text
          style={
            Object {
              "color": "#202452",
              "fontSize": 10,
              "fontWeight": "500",
              "marginTop": 2,
            }
          }
        >
          03 January, 2022
        </Text>
      </View>
    </View>
    <Text
      style={
        Object {
          "color": "#202452",
          "fontSize": 17,
          "fontWeight": "bold",
          "paddingLeft": 110,
        }
      }
    >
      2. Visits
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#202452",
        "borderBottomLeftRadius": 26,
        "flex": 1,
        "flexDirection": "row",
        "height": "100%",
        "justifyContent": "center",
        "left": "80.4%",
        "position": "absolute",
        "width": "20%",
        "zIndex": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#FFFFFF",
          "fontSize": 17,
          "fontWeight": "bold",
          "marginRight": 90,
        }
      }
    >
      3. Routes
    </Text>
  </View>
</View>
`;
